File: 6-zigzag-conversion/6-zigzag-conversion.cpp
class Solution {
public:
    string convert(string s, int numRows) {
        string ans;
        if(numRows<2){
            return s;
        }
        for(int i=0;i<numRows;i++){
            int k=i;
            int g=1;
            for(int j=0;j<s.size() && k<s.size();j++){
                if(g!=0)
                    ans.push_back(s[k]);
                // cout<<s[k]<<","<<k<<endl;
                if(j%2==0){
                    g=2*(numRows-i-1);
                    k+=g;
                }
                else{
                    g=2*i;
                    k+=g;
                }
            }
        }
        return ans;
    }
};

File: 6-zigzag-conversion/6-zigzag-conversion.py
class Solution:
    def convert(self, s: str, numRows: int) -> str:
        if(numRows<2):
            return s
        ans=[]
        for i in range(numRows):
            k=i
            gap=1
            bit=True
            while(k<len(s)):
                if(gap!=0):
                    ans.append(s[k])
                if(bit):
                    gap=2*(numRows-i-1)
                else:
                    gap=2*i   
                k+=gap
                bit=not bit
        return "".join(ans)
                